src/compiler/transformers/module/module.ts

```diff
@@ -604,16 +626,91 @@ namespace ts {
+            return factory.updateExpressionStatement(
+                node,
+                visitNode(node.expression, discardedValueVisitor, isExpression)
+            );
+        }
+
+        function visitParenthesizedExpression(node: ParenthesizedExpression, valueIsDiscarded: boolean) {
+            return factory.updateParenthesizedExpression(node, visitNode(node.expression, valueIsDiscarded ? discardedValueVisitor : visitor, isExpression));
+        }
+
+        function visitPartiallyEmittedExpression(node: PartiallyEmittedExpression, valueIsDiscarded: boolean) {
+            return factory.updatePartiallyEmittedExpression(node, visitNode(node.expression, valueIsDiscarded ? discardedValueVisitor : visitor, isExpression));
+        }
+
+        function visitPreOrPostfixUnaryExpression(node: PrefixUnaryExpression | PostfixUnaryExpression, valueIsDiscarded: boolean) {
+            // When we see a prefix or postfix increment expression whose operand is an exported
+            // symbol, we should ensure all exports of that symbol are updated with the correct
+            // value.
+            //
+            // - We do not transform generated identifiers for any reason.
+            // - We do not transform identifiers tagged with the LocalName flag.
+            // - We do not transform identifiers that were originally the name of an enum or
+            //   namespace due to how they are transformed in TypeScript.
+            // - We only transform identifiers that are exported at the top level.
+            if ((node.operator === SyntaxKind.PlusPlusToken || node.operator === SyntaxKind.MinusMinusToken)
+                && isIdentifier(node.operand)
+                && !isGeneratedIdentifier(node.operand)
+                && !isLocalName(node.operand)
+                && !isDeclarationNameOfEnumOrNamespace(node.operand)) {
+                const exportedNames = getExports(node.operand);
+                if (exportedNames) {
+                    let temp: Identifier | undefined;
+                    let expression: Expression = visitNode(node.operand, visitor, isExpression);
+                    if (isPrefixUnaryExpression(node)) {
+                        expression = factory.updatePrefixUnaryExpression(node, expression);
+                    }
+                    else {
+                        expression = factory.updatePostfixUnaryExpression(node, expression);
+                        if (!valueIsDiscarded) {
+                            temp = factory.createTempVariable(hoistVariableDeclaration);
+                            expression = factory.createAssignment(temp, expression);
+                            setTextRange(expression, node);
+                        }
+                        expression = factory.createComma(expression, factory.cloneNode(node.operand));
+                        setTextRange(expression, node);
+                    }
+
+                    for (const exportName of exportedNames) {

```

Comment:
I don't understand what this loop is for. Is it doing something that the code path through visitEachChild would normally do?

Reply:
Whenever a variable is exported using `export {}` it will have one or more additional exported names:

```ts
// source
// @target: esnext
// @module: commonjs
let x = 1;
x++;
export { x as a };
export { x as b };

// output
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.b = exports.a = void 0;
let x = 1;
exports.a = x;
exports.b = x;
exports.b = exports.a = (x++, x);
```

In the above case, `exportedNames` would contain `a` and `b`, so every time we change the value of `x` we also need to update the exports for `a` and `b`.